landlab/components/space/space.py

```diff
@@ -295,14 +335,29 @@ def __init__(self, grid, K_sed=None, K_br=None, F_f=None,
             self.discharge_field = None
             
         if self.method == 'simple_stream_power':
-            self.simple_stream_power()
+            self.calc_ero_rates = self.simple_stream_power
         elif self.method == 'threshold_stream_power':
-            self.threshold_stream_power()
+            self.calc_ero_rates = self.threshold_stream_power
         elif self.method == 'stochastic_hydrology':
-            self.stochastic_hydrology()
+            self.calc_ero_rates = self.stochastic_hydrology
         else:
+            print('METHOD:')
+            print(self.method)
             raise ValueError('Specify erosion method (simple stream power,\
                             threshold stream power, or stochastic hydrology)!')
+
+        # Handle option for solver
+        if solver == 'basic':
+            self.run_one_step = self.run_one_step_basic
+        elif solver == 'celerity':
+            self.run_one_step = self.run_with_simple_time_step_adjuster
+        elif solver == 'adaptive':
+            self.run_one_step = self.run_with_adaptive_time_step_solver
+            self.time_to_flat = np.zeros(grid.number_of_nodes)
+        else:
+            raise ValueError("Parameter 'solver' must be one of: "
+                             + "'basic', 'adaptive', 'celerity'")
+
```

Comment:
Otherwise, it should be ready to use, with the caveat that it hasn't been tested against analytical solutions yet. Charlie has agreed to do that today or tomorrow.